### PR TITLE
TL #258 - Export XML

### PR DESCRIPTION
**Pull request by dleadbetter**
This pull request fixes a bug where exporting a project to XML would order the transcriptions incorrectly. The solution was to refactor the `tei_xml` method to recursively traverse the `document_node` records for the selected document, order by the `position`. Resulting file [here](https://github.com/performant-software/textlab/files/8141573/121.xml.txt).

### CODE REVIEW

File: app/models/document.rb [hunk 1]
```diff
@@ -31,43 +31,27 @@ def self.get_all( current_user )
...
-            transcription = Transcription.find_by(leaf_id: child.leaf_id)
-            if transcription.present?
-              if transcription.content.present?
-                xml_string = "#{xml_string}#{transcription.content}"
-              end
-            end
-          end
-        end
-      end
+  def append_node(node, xml)
```
Comment (dleadbetter) [inline app/models/document.rb:37]: This is the only real change, everything else is whitespace. Using the "Hide whitespace" option is helpful.

**Review by NickLaiacona — COMMENTED** [review]
Looking at the output file, it gets to #img_198 and stops. However, the last published image is in the 800s.

**Comment by dleadbetter** [comment]
> Looking at the output file, it gets to #img_198 and stops. However, the last published image is in the 800s.

It looks like "leaf 198" contains the last published transcription. It is nested under Inscribed Versos > InscribedVersosCh7. Above the "leaf 198" transcription contains transcripts with images in the 800s (#img_810-0020, #img_810-0019, etc), which I believe appears to be working as expected.

**Comment by NickLaiacona** [comment]
Ah, that's great. OK, otherwise looks good.

**Pull request merged**